wolfssl/wolfcrypt/types.h

```diff
@@ -316,6 +316,16 @@ typedef struct w64wrapper {
         #endif
     #endif /* WC_MAYBE_UNUSED */
 
+    #ifndef WC_DO_NOTHING
+        #define WC_DO_NOTHING do {} while (0)
+        #ifdef _MSC_VER
+            /* disable buggy MSC warning around while(0),
+             *"warning C4127: conditional expression is constant"
+             */
+            #pragma warning(disable: 4127)
```

Comment:
Should this warning be disabled globally or only for WC_DO_NOTHING?  Maybe consider pragma push, pragma disable, do-while, pragma pop?

Reply:
it's incumbent code from years ago -- didn't want to overturn the applecart (one-thing-at-a-time).